src/catchup/CatchupManagerImpl.cpp

```diff
@@ -181,12 +187,21 @@ CatchupManagerImpl::processLedger(LedgerCloseData const& ledgerData)
         hm.isFirstLedgerInCheckpoint(firstLedgerInBuffer) &&
         firstLedgerInBuffer < lastLedgerInBuffer)
     {
-        message =
-            fmt::format(FMT_STRING("Starting catchup after ensuring checkpoint "
-                                   "ledger {:d} was closed on network"),
-                        lastLedgerInBuffer);
-
-        startOnlineCatchup();
+        // No point in processing ledgers as catchup won't ever be able to
+        // succeed
+        if (!mCatchupFatalFailure)
+        {
+            message = fmt::format(
+                FMT_STRING("Starting catchup after ensuring checkpoint "
+                           "ledger {:d} was closed on network"),
+                lastLedgerInBuffer);
+            startOnlineCatchup();
+        }
+        else
+        {
+            CLOG_FATAL(History, "Skipping catchup: incompatible core version "
```

Comment:
This is interesting -- core will just loop forever at this point right? Should we notice at some higher level and shut down the node with an error? I mean the user might still have core restart endlessly but maybe they're more likely to notice that than just endless LOG_FATALs?

Reply:
That's fair, though if we shut down the node and restart it, we'll go through the catchup flow again, since we'll lose any in-memory state. So maybe we just keep it as-is, wdyt?

Reply:
yeah, fair point!